gradle/setup.gradle

```diff
@@ -37,6 +37,12 @@ task javadocJar(type: Jar) {
 }
 
 publishing {
+    repositories {
```

Comment:
@melix just curious, why not already available `publishToMavenLocal`? Or would you recommend _against_ using it?

/cc @simonbasle

Reply:
I'm fighting hard to avoid using `mavenLocal`: it's a trashcan where every Maven project publishes things locally. Having this project-local repository makes it much easier to figure out what your project generates.

Reply:
Since `mavenLocal` is a pre-configured task in Gradle (unlike the mock repo), I guess the fight isn't over yet? :)

My suggestion is to not add a custom publication (that needs to be documented as well) and wait for Gradle to add it as a pre-configured task (if Gradle decides to deprecate & remove `publishToMavenLocal`).

@simonbasle WDYT?

Reply:
I think there's no harm in keeping it as in the PR.

we're not talking about a separate `publication` but a `repository`, so it should have no impact on more "standard" Gradle tasks right @melix ?

Plus `publishToMavenLocal()` does have legit use (eg. making a gradle-built artifact available to other projects built on the same CI, but with Maven) so it's not going to go anywhere (I guess).

Reply:
That's correct, I see no reason not to keep it, it just adds a repository to deploy to, if you want to. And yes, `publishToMavenLocal()` makes sense in some cases like interaction with Maven.